src/core.jl

```diff
@@ -711,4 +711,31 @@ Base.fill!(A::DArray, x) = begin
         @async remotecall_fetch((A,x)->(fill!(localpart(A), x); nothing), p, A, x)
     end
     return A
+end
+
+# new work #
+
+function star(f1,f2,f3,args::DistributedArrays.DArray...)
+        let ds = ds
+            out1 = [remotecall( ()-> f1( DistributedArrays.localpart(ds),2 )[end],p ) for p = workers()]
+
+            out2 = [0; f2(map(fetch,out1))]
```

Comment:
The `0` prepending shouldn't be part of the `star` framework but rather part of `f2`.